src/main/java/org/candlepin/model/ContentCurator.java

```diff
@@ -645,63 +646,65 @@ public Set<Content> getChildrenContentOfProductsByUuids(Collection<String> produ
+                    UNION
+                    SELECT CAST(product_uuid AS CHAR(32)), CAST(provided_product_uuid AS CHAR(32))
+                    FROM cp_product_provided_products
+                ),
+                product_graph(uuid, depth) AS (
+                    SELECT CAST(pool.product_uuid AS CHAR(32)), 0
+                    FROM cp_pool pool
+                    WHERE pool.owner_id = :owner_id
+                    AND pool.startDate <= :start_date AND pool.endDate >= :end_date
+                    UNION
+                    SELECT CAST(pcmap.child_uuid AS CHAR(32)), pgraph.depth + 1
+                    FROM parent_child_map pcmap
+                    JOIN product_graph pgraph ON pgraph.uuid = pcmap.parent_uuid
+                    WHERE pgraph.depth < 10
+                )
+                SELECT content.uuid, pc.enabled
+                FROM product_graph pgraph
+                JOIN cp_product_contents pc ON pc.product_uuid = pgraph.uuid
+                JOIN cp_contents content ON content.uuid = pc.content_uuid
+            """;
+
+        Map<String, Boolean> contentEnablementMap =
+            (Map<String, Boolean>) entityManager.createNativeQuery(sql)
             .setParameter("owner_id", ownerId)
             .setParameter("start_date", now)
             .setParameter("end_date", now)
-            .getResultList());
-
-        Query prodQuery = entityManager.createNativeQuery(sql);
-        int blockSize = Math.min(this.getInBlockSize(), this.getQueryParameterLimit() / 2);
-
-        // Get all of the derived products and provided products for the products we've found, recursively
-        for (Set<String> lastBlock = productUuids; !lastBlock.isEmpty();) {
-            Set<String> children = new HashSet<>();
-
-            for (List<String> block : this.partition(lastBlock, blockSize)) {
-                children.addAll(prodQuery.setParameter("product_uuids", block)
-                    .getResultList());
-            }
-
-            productUuids.addAll(children);
-            lastBlock = children;
-        }
-
-        // Use the product UUIDs to select all related enabled content
-        jpql = "SELECT DISTINCT pc.content, pc.enabled FROM ProductContent pc " +
-            "WHERE pc.product.uuid IN (:product_uuids)";
-
-        Query contentQuery = entityManager.createQuery(jpql);
-        HashMap<Content, Boolean> activeContent = new HashMap<>();
-
-        for (List<String> block : this.partition(productUuids)) {
-            contentQuery.setParameter("product_uuids", block)
-                .getResultList()
-                .forEach(col -> {
-                    Content content = (Content) ((Object[]) col)[0];
-                    Boolean enabled = (Boolean) ((Object[]) col)[1];
-                    activeContent.merge(content, enabled, (v1, v2) ->
-                        v1 != null && v1.booleanValue() ? v1 : v2);
-                });
+            .getResultList()
+            .stream()
```

Comment:
You should be able to call `.getResultStream()` here instead of `.getResultList().stream()`.

Reply:
i was looking for that, but for that you need JPA api 2.2 and we are using 2.1 so we need to use this option, or maybe I missed something.

Reply:
We can bump the API version to 2.2 without running into the package name change:

`hibernate = { module = "javax.persistence:javax.persistence-api", version = "2.2" }`

Reply:
done